src/test_l1_tvSettings.c

```diff
@@ -17615,6 +17617,121 @@ void test_l1_tvSettings_negative_GetLdimZoneShortCircuitStatus (void)
+ */
+void test_l1_tvSettings_positive_GetNumberOfDimmingZones (void)
+{
+	gTestID = 227;   /* It must be 227 */
+	UT_LOG("In:%s [%02d%03d]", __FUNCTION__,gTestGroup,gTestID);
+
+    tvError_t result = tvERROR_NONE ;
+    unsigned int number_of_dimming_zones = 0;
+    unsigned int number_of_dimming_zones_retry = 0;
+
+    /* Step 01: Calling tvsettings initialization and expecting the API to return success */
+    result = TvInit();
+    UT_ASSERT_EQUAL_FATAL(result, tvERROR_NONE);
+
+    /* Step 02: Calling tvsettings GetNumberOfDimmingZones and the API to return success */
+    result = GetNumberOfDimmingZones(&number_of_dimming_zones);
+    if(UT_KVP_PROFILE_GET_BOOL("tvSettings/LDIMShortCircuitStatus/enable") == true) {
+        UT_ASSERT_EQUAL(result, tvERROR_NONE);
+    } else {
+        UT_ASSERT_EQUAL(result, tvERROR_OPERATION_NOT_SUPPORTED);
+    }
+
+    /* Step 03: Retry tvsettings GetNumberOfDimmingZones and the API to return success */
+    result = GetNumberOfDimmingZones(&number_of_dimming_zones_retry);
+    if(UT_KVP_PROFILE_GET_BOOL("tvSettings/LDIMShortCircuitStatus/enable") == true) {
+        UT_ASSERT_EQUAL(result, tvERROR_NONE);
+    } else {
+        UT_ASSERT_EQUAL(result, tvERROR_OPERATION_NOT_SUPPORTED);
+    }
+
+    UT_ASSERT_EQUAL(number_of_dimming_zones_retry, number_of_dimming_zones);
+
+    /* Step 04: Calling tvsettings termination and expecting the API to return success */
+    result = TvTerm();
+    UT_ASSERT_EQUAL_FATAL(result, tvERROR_NONE);
+
+    UT_LOG("Out %s",__FUNCTION__);
+}
+
+/**
+ * @brief Validate GetNumberOfDimmingZones() for all negative invocation scenarios
+ *
+ * **Test Group ID:** Basic : 01@n
+ * **Test Case ID:** 228@n
+ *
+ * **Pre-Conditions:** None@n
+ *
+ * **Dependencies:** None@n
+ * **User Interaction:** None
+ *
+ * **Test Procedure:**@n
+ * | Variation / Step | Description | Test Data | Expected Result | Notes |
+ * | :-------: | ------------- | --------- | --------------- | ----- |
+ * | 01 | call GetNumberOfDimmingZones() -  GetNumberOfDimmingZones Status before TvInit() | unsigned int*  | tvERROR_INVALID_STATE | Should Pass |
+ * | 02 | call TvInit() -  Initialise and get a valid instance of the TV client | void | tvERROR_NONE | Should Pass |
+ * | 03 | call GetNumberOfDimmingZones() -  GetNumberOfDimmingZones Status with invalid argument and | NULL | tvERROR_INVALID_PARAM | Should Pass |
+ * | 04 | call TvTerm() -  Terminate and close the instance of the TV client | void | tvERROR_NONE | Should Pass |
+ * | 05 | call GetNumberOfDimmingZones() -  GetNumberOfDimmingZones Status with valid argument | unsigned int* | tvERROR_INVALID_STATE | Should Pass |
+ */
+void test_l1_tvSettings_negative_GetNumberOfDimmingZones (void)
+{
+	gTestID = 228;   /* It must be 228 */
+	UT_LOG("In:%s [%02d%03d]", __FUNCTION__,gTestGroup,gTestID);
+
+    tvError_t result = tvERROR_NONE ;
+    unsigned int number_of_dimming_zones = 0;
+
+    if (extendedEnumsSupported == true)
```

Comment:
extendedEnumsSupported is wrongly implemented in this module 
let's take this has separate issues

Reply:
Sorry, couldn't get it.  Could you please explain it ?